pkg/services/alerting/rule.go

```diff
@@ -136,14 +137,14 @@ func NewRuleFromDBAlert(ruleDef *models.Alert) (*Rule, error) {
 
 	for _, v := range ruleDef.Settings.Get("notifications").MustArray() {
 		jsonModel := simplejson.NewFromAny(v)
-		if id, err := jsonModel.Get("id").Int64(); err == nil {
-			model.Notifications = append(model.Notifications, fmt.Sprintf("%09d", id))
-		} else {
-			uid, err := jsonModel.Get("uid").String()
-			if err != nil {
-				return nil, ValidationError{Reason: "Neither id nor uid is specified in 'notifications' block, " + err.Error(), DashboardID: model.DashboardID, AlertID: model.ID, PanelID: model.PanelID}
-			}
+		if uid, err := getNotificationUidFromIdField(jsonModel, ruleDef); err == nil {
+			model.Notifications = append(model.Notifications, uid)
+		} else if err.Error() != "invalid value type" {
+			return nil, ValidationError{Reason: "Unable to translate notification id to uid, " + err.Error(), DashboardID: model.DashboardID, AlertID: model.ID, PanelID: model.PanelID}
+		} else if uid, err := getNotificationUidFromUidField(jsonModel); err == nil {
 			model.Notifications = append(model.Notifications, uid)
+		} else {
+			return nil, ValidationError{Reason: "Neither id nor uid is specified in 'notifications' block, " + err.Error(), DashboardID: model.DashboardID, AlertID: model.ID, PanelID: model.PanelID}
```

Comment:
I think that this block was more readable as it was before.
Therefore I would suggest to get rid of `getNotificationUidFromUidField` and `getNotificationUidFromIdField` and change it to:
```
if id, err := jsonModel.Get("id").Int64(); err == nil {
    uid, err := translateNotificationIDToUID(id, ruleDef.OrgId)
    if err != nil {
        return nil, ValidationError{Reason: "Unable to translate notification id to uid, " + err.Error(), DashboardID: model.DashboardID, AlertID: model.ID, PanelID: model.PanelID}
    }
    model.Notifications = append(model.Notifications, uid)
} else {
    uid, err := jsonModel.Get("uid").String()
    if err != nil {
        return nil, ValidationError{Reason: "Neither id nor uid is specified in 'notifications' block, " + err.Error(), DashboardID: model.DashboardID, AlertID: model.ID, PanelID: model.PanelID}
    }
    model.Notifications = append(model.Notifications, uid)
}
```

Reply:
sure, i inlined it :)